File: core/src/com/questmark/input/KeyInput.java
package com.questmark.input;

import com.badlogic.gdx.Input;
import com.badlogic.gdx.InputProcessor;

/**
 * Key input handler for the game that processes key input for entity movement, skills, etc.
 *
 * @author Ming Li
 */
public class KeyInput implements InputProcessor {

    private KeyInputHandler keyInputHandler;

    public KeyInput(KeyInputHandler keyInputHandler) {
        this.keyInputHandler = keyInputHandler;
    }

    @Override
    public boolean keyDown(int keycode) {
        switch (keycode) {
            case Input.Keys.W:
                keyInputHandler.move(Direction.Up);
                break;
            case Input.Keys.S:
                keyInputHandler.move(Direction.Down);
                break;
            case Input.Keys.A:
                keyInputHandler.move(Direction.Left);
                break;
            case Input.Keys.D:
                keyInputHandler.move(Direction.Right);
                break;
        }
        return true;
    }

    @Override
    public boolean keyUp(int keycode) {
        switch (keycode) {
            case Input.Keys.W:
                keyInputHandler.stop(Direction.Up);
                break;
            case Input.Keys.S:
                keyInputHandler.stop(Direction.Down);
                break;
            case Input.Keys.A:
                keyInputHandler.stop(Direction.Left);
                break;
            case Input.Keys.D:
                keyInputHandler.stop(Direction.Right);
                break;
        }
        return true;
    }

    @Override
    public boolean keyTyped(char character) {
        return true;
    }

    /** The key event listener methods after this don't matter as much **/

    @Override
    public boolean touchDown(int screenX, int screenY, int pointer, int button) {
        return false;
    }

    @Override
    public boolean touchUp(int screenX, int screenY, int pointer, int button) {
        return false;
    }

    @Override
    public boolean touchDragged(int screenX, int screenY, int pointer) {
        return false;
    }

    @Override
    public boolean mouseMoved(int screenX, int screenY) {
        return false;
    }

    @Override
    public boolean scrolled(int amount) {
        return false;
    }

}

File: core/src/com/questmark/input/KeyInput.kt
package com.questmark.input

import com.badlogic.gdx.Input
import com.badlogic.gdx.InputProcessor

/**
 * Key input handler for the game that processes key input for entity movement, skills, etc.
 *
 * @author Ming Li
 */
class KeyInput(private val keyInputHandler: KeyInputHandler) : InputProcessor {

    override fun keyDown(keycode: Int): Boolean {
        when (keycode) {
            Input.Keys.W -> keyInputHandler.move(Direction.Up)
            Input.Keys.S -> keyInputHandler.move(Direction.Down)
            Input.Keys.A -> keyInputHandler.move(Direction.Left)
            Input.Keys.D -> keyInputHandler.move(Direction.Right)
        }
        return true
    }

    override fun keyUp(keycode: Int): Boolean {
        when (keycode) {
            Input.Keys.W -> keyInputHandler.stop(Direction.Up)
            Input.Keys.S -> keyInputHandler.stop(Direction.Down)
            Input.Keys.A -> keyInputHandler.stop(Direction.Left)
            Input.Keys.D -> keyInputHandler.stop(Direction.Right)
        }
        return true
    }

    override fun keyTyped(character: Char): Boolean {
        return true
    }

    /** The key event listener methods after this don't matter as much  */

    override fun touchDown(screenX: Int, screenY: Int, pointer: Int, button: Int): Boolean {
        return false
    }

    override fun touchUp(screenX: Int, screenY: Int, pointer: Int, button: Int): Boolean {
        return false
    }

    override fun touchDragged(screenX: Int, screenY: Int, pointer: Int): Boolean {
        return false
    }

    override fun mouseMoved(screenX: Int, screenY: Int): Boolean {
        return false
    }

    override fun scrolled(amount: Int): Boolean {
        return false
    }

}
